src/Tokenizer/Tokens.php

```diff
@@ -869,31 +869,61 @@ public function findSequence(array $sequence, $start = 0, $end = null, $caseSens
     public function insertAt($index, $items)
     {
         $items = \is_array($items) || $items instanceof self ? $items : [$items];
-        $itemsCnt = \count($items);
 
-        if (0 === $itemsCnt) {
+        $this->insertSlices([$index => $items]);
+    }
+
+    /**
+     * Insert a slices or individual Tokens into multiple places in a single run.
+     *
+     * @internal
+     *
+     * @param array<int, Token|Token[]> $slices
```

Comment:
What I'm missing in this doc is the fact that this `int` key is used as insert-index, while it is mentioned in for example `insertAt`. Leaving it out here may cause one to think that it's not the same idea.

Also, isn't `Tokens` also accepted, thus shouldn't the typehint be `array<int, Token|Token[]|Tokens>`?

Reply:
:+1: